Add OCSP_RESPONSE_check_status(), a function to check OCSP response for revoked certificate in s_client.

OttoHollmann opened the pull request:
During our audit of OpenSSL we found that `openssl s_client` correctly identify revoked certificate in OCSP response (print information about certificate status - e.g. revoked) but it doesn't terminate connection.

According to documentation https://www.openssl.org/docs/manmaster/man3/SSL_CTX_set_tlsext_status_arg.html it's bug/missing check in callback function.

```
The callback when used on the client side should return a negative value
on error; 0 if the response is not acceptable (in which case the handshake
will fail) or a positive value if it is acceptable.
```

So I'm adding new function `OCSP_RESPONSE_check_status` that will check OCSP response and return 0 in case of revoked certificate (according to above documentation). Also I added this function into `ocsp_resp_cb()` that was set as a callback function via `SSL_CTX_set_tlsext_status_cb` in s_client.



This issue also affects 1.1.1 branch so please let me know if I should prepare PR for other branches or keep it only in master branch.

Documentation is missing, as well as for other OCSP functions, but I can document this one if you want.

##### Checklist
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->
- [ ] documentation is added or updated
- [ ] tests are added or updated

t8m commented:
This is a new functionality added to s_client which also requires a new public API call so clearly a feature. This IMO needs a testcase (either direct or via the s_client invocation) and the function call needs a documentation.

OttoHollmann commented:
I rebased branch and added test and documentation.
Unfortunately I don't know how to determine in perl if OCSP responder is ready so that's the reason why there is no check.

openssl-machine commented:
This PR is in a state where it requires action by @openssl/otc but the last update was 30 days ago

openssl-machine commented:
This PR is in a state where it requires action by @openssl/otc but the last update was 61 days ago

openssl-machine commented:
This PR is in a state where it requires action by @openssl/otc but the last update was 92 days ago

openssl-machine commented:
This PR is in a state where it requires action by @openssl/otc but the last update was 123 days ago

openssl-machine commented:
This PR is in a state where it requires action by @openssl/otc but the last update was 154 days ago

openssl-machine commented:
This PR is in a state where it requires action by @openssl/otc but the last update was 185 days ago

openssl-machine commented:
This PR is in a state where it requires action by @openssl/otc but the last update was 216 days ago

openssl-machine commented:
This PR is in a state where it requires action by @openssl/otc but the last update was 247 days ago

openssl-machine commented:
This PR is in a state where it requires action by @openssl/otc but the last update was 278 days ago

nhorman commented:
This pr.has.been.idle.for over a year.  Given the lack of input with failing ci checks, I'm closing it.  Please reopen if there is interest in continuing this work
